common/web/keyboard-processor/build.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+#
+# Compile KeymanWeb's 'keyboard-processor' module, one of the components of Web's 'core' module.
+#
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../resources/build/build-utils.sh"
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+# This script runs from its own folder
+cd "$(dirname "$THIS_SCRIPT")"
+
+action=
+function action_failure() {
+  if [ -n "$action" ]; then
+    builder_report failure $action
+  fi
+}
+
+trap action_failure err
```

Comment:
I think it'd be good to move this to build-utils.sh and make it a standard part of the build, in which case perhaps use `_builder_current_action` as the variable for tracking. We'd set that within the `builder_has_action` call.

(Separate note, I want to rename `builder_has_action` to `builder_start_action`, and create a new side-effect-free `builder_has_action` which doesn't emit the "starting action" message or set this var. This has been on my mind for a little while, but don't feel like you need to do this here -- you'd need to fixup all the other scripts which call `builder_has_action`!)

We'd enable this trap in `builder_describe()` -- that way it doesn't impact existing scripts.